lib/templates/joins.genie

```diff
@@ -0,0 +1,86 @@
+    ) => C.makeSelfJoinFromParam($loader, table, param);
+
+    ( 'paris of' table:short_phrase_table ('such that'|'that') pronoun_the_first ('has'|'have'|'had') condition:npp_join_condition
+    | 'paris of' table:short_phrase_table ('such that'|'that') pronoun_the_first ('is'|'are'|'was'|'were') ('the'|'a'|'') condition:npi_join_condition
+    | 'paris of' table:short_phrase_table ('such that'|'that') pronoun_the_first ('is'|'are'|'was'|'were') condition:pvp_join_condition 
+    | 'paris of' table:short_phrase_table ('such that'|'that') pronoun_the_first ('is'|'are'|'was'|'were') condition:preposition_join_condition 
+    | 'paris of' table:short_phrase_table ('such that'|'that') pronoun_the_first condition:avp_join_condition 
+    | 'pairs of' table:short_phrase_table ('such that'|'that') pronoun_the_first condition:reverse_verb_join_condition 
+    | table:short_phrase_table 'pairs' ('such that'|'that') pronoun_the_first ('has'|'have'|'had') condition:npp_join_condition
+    | table:short_phrase_table 'pairs' ('such that'|'that') pronoun_the_first ('is'|'are'|'was'|'were') ('the'|'a'|'') condition:npi_join_condition
+    | table:short_phrase_table 'pairs' ('such that'|'that') pronoun_the_first ('is'|'are'|'was'|'were') condition:pvp_join_condition 
+    | table:short_phrase_table 'pairs' ('such that'|'that') pronoun_the_first ('is'|'are'|'was'|'were') condition:preposition_join_condition 
+    | table:short_phrase_table 'pairs' ('such that'|'that') pronoun_the_first condition:avp_join_condition 
+    | table:short_phrase_table 'pairs' ('such that'|'that') pronoun_the_first condition:reverse_verb_join_condition 
+    ) => C.makeSelfJoin(table, condition);
+
+    table:short_phrase_table ('pairs'|'') ('that'|'which'|'who') ('are'|'were') ('the'|'') param:out_param_Any with { symmetric = true } 'of' each_other
+    => C.makeSelfJoinFromParam($loader, table, param);
+
+    ( table:short_phrase_table ('pairs'|'') ('that'|'which'|'who') ('are'|'were')  ('the'|'') condition:npi_symmetric_join_condition 
+    | table:short_phrase_table ('pairs'|'') ('that'|'which'|'who') ('are'|'were') condition:pvp_symmetric_join_condition 
+    | table:short_phrase_table ('pairs'|'') ('that'|'which'|'who') ('are'|'were') condition:preposition_symmetric_join_condition 
+    | table:short_phrase_table ('pairs'|'') ('that'|'which'|'who') condition:avp_symmetric_join_condition 
+    | table:short_phrase_table ('pairs'|'') ('that'|'which'|'who') condition:reverse_verb_symmetric_join_condition
+    ) => C.makeSelfJoin(table, condition);
+}
+
+generic_join_table = { 
+    table1:short_phrase_table ('and'|'with') 'the' table2:base_table_hidden param2:out_param_Any 'of' ('its'|'their') param1:out_param_Any
+    => C.makeGenericJoin($loader, table1, param1, table2, param2);
+}
+
+complete_table : Ast.Expression = {
+    self_join_table;
+    generic_join_table;
```

Comment:
Please gate these under "notablejoin" so I can disable them in production and multiwoz